src/util/date-helper.ts

```diff
@@ -0,0 +1,68 @@
+export namespace DateHelper {
+    export const clampDay = (
+        day: string,
+        month: string,
+        year: string
+    ): string => {
+        const dayNumber = parseInt(day);
+        const monthNumber = parseInt(month);
+        const yearNumber = parseInt(year);
+
+        if (dayNumber == 0) {
+            return "1";
+        }
+
+        switch (monthNumber) {
```

Comment:
suggestion: can try squishing the code this way, keeping the return values together

pseudo code (using if-else for brevity, but switch is fine too)
```
if (months_with_30_days.includes(month)) {
    return Math.min(30, day)
}
if (months_with_31_days.includes(month)) {
    return Math.min(31, day)
}
if (feb) {
    return leapYear ? Math.min(28, day) : Math.min(29, day)
}
return day
```